model/src/cmd/cmdmodel.rb.in

```diff
@@ -0,0 +1,169 @@
+  require 'dl/import'
+  include DL
+else
+  require 'fiddle'
+  require 'fiddle/import'
+  include Fiddle
+end
+
+require 'date'
+require 'optparse'
+
+# Constants.
+LIBRARY_NAME = '@model_library_location@'
+LIBRARY_VERSION = '@PROJECT_VERSION_FULL@'
+COMMON_OPTIONS =
+  "  -h [ --help ]              Print this help message.                   \n"\
+  "                                                                        \n"\
+
+COMMANDS = { 'model' =>
+  "Print information about models.\n\n"+
+  "                                                                        \n"\
+  "  ign model [options]                                                   \n"\
+  "                                                                        \n"\
+  "Available Options:                                                      \n"\
+  "  --list                     Get a list of the available models.        \n"\
+  "  -m [--model] arg           Select the model to be shown.              \n"\
+  "  -l [--link]  arg           Select a link to show its properties.      \n"\
+  "                             If no arg is passed all links are printed  \n"\
+  "                             Requires the -m option                     \n"\
+  "                                                                        \n"\
+  "  -j [--joint] arg           Select a joint to show its properties.     \n"\
+  "                             If no arg is passed all joints are printed \n"\
+  "                             Requires the -m option                     \n"\
+  "                                                                        \n"\
+  "                             E.g. to get information about the          \n"\
+  "                             box_link in the shapes world, run:         \n"\
+  "                             ign model -m box -l box_link               \n\n"+
+  COMMON_OPTIONS
+}
+
+#
+# Class for the Ignition Gazebo Model command line tools.
+#
+class Cmd
+
+  #
+  # Return a structure describing the options.
+  #
+  def parse(args)
+    options = {
+      'pose' => 0,
+      'link' => 0,
+      'link_name' => '',
+      'joint' => 0,
+      'joint_name' => ''
+    }
+    usage = COMMANDS[args[0]]
+
+    opt_parser = OptionParser.new do |opts|
+      opts.banner = usage
+  
+      opts.on('-h', '--help') do
+          puts usage
+          exit
+      end
+      opts.on('-m', '--model [arg]', String, 'Model name') do |m|
+        options['model'] = m
+      end
+      opts.on('--list', String, 'List models available') do
+        options['list'] = 1
+      end
+      opts.on('-p', '--pose', Integer, 'Request pose') do
+        options['pose'] = 1
+      end
+      opts.on('-l', '--link [arg]', String, 'Request link information') do |l|
+        options['link'] = 1
+        options['link_name'] = l
+      end
+      opts.on('-j', '--joint [arg]', String, 'Request joint information') do |j|
+        options['joint'] = 1
+        options['joint_name'] = j
```

Comment:
Is it necessary to keep both the boolean `joint` and the string `joint_name`? I think the boolean can be removed and we can check if the string is empty before using it.

See for example how we handle the `--record-path` option:

https://github.com/ignitionrobotics/ign-gazebo/blob/f2706be7415fc267da4fb140b8e3b4549781f1b7/src/cmd/ign.cc#L70

Reply:
Addressed, thanks for the example!